### PR TITLE
Monasca

### PR DESCRIPTION
**Pull request by sjamgade**
incremental pull request

### CODE REVIEW

**Comment by jgrassler** [comment]
Note: for now we are targeting `stable/newton` with this since this is where we need to get this running first.

File: scripts/openstack-quickstart-demosetup [hunk 1]
```diff
@@ -787,6 +790,36 @@ if [ "x$with_gnocchi" = "xyes" ]; then
     #start_and_enable_service openstack-gnocchi-statsd
 fi
 
+#-----------------------------------------
+# setup Monasca
+#-----------------------------------------
+
+if [ "x$with_monasca" = "xyes" ]; then
+# monasca-log-agent
+    cp /etc/monasca-log-agent/agent.conf.sample /etc/monasca-log-agent/agent.conf
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: this should be done in packaging instead..

File: scripts/openstack-quickstart-demosetup [hunk 2]
```diff
@@ -787,6 +790,36 @@ if [ "x$with_gnocchi" = "xyes" ]; then
...
+    c="/etc/monasca-log-agent/agent.conf"
+    #TODO(sjamgade): change these the following two ips to $IP
+    sed -i "s:\(monasca_log_api_url =>\).*:\1 \"http\://$MONASCA_HOST\:8070/v3.0\":" $c
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: why not use crudini?

File: scripts/openstack-quickstart-demosetup [hunk 3]
```diff
@@ -787,6 +790,36 @@ if [ "x$with_gnocchi" = "xyes" ]; then
...
+    sed -i "s:\(dimensions => \).*:\1 [ \"hostname\:monasca\" ]:" $c
+    start_and_enable_service  openstack-monasca-log-agent
+
+# monasca-log-agent
+    monasca-setup \
+    -u 'admin' \
+    -p '$pw' \
+    --skip_enable \
+    --user 'monasca-agent' \
+    --keystone_url "http://$IP:5000/v3/" \
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: I think we have a variable that contains the keystone endpoint somewhere else already.. lets make use of that one

File: scripts/openstack-quickstart-demosetup [hunk 4]
```diff
@@ -787,6 +790,36 @@ if [ "x$with_gnocchi" = "xyes" ]; then
...
+    sed -i "s:\(keystone_api_url =>\).*:\1 \"http\://$IP\:5000/v3/\":" $c
+    sed -i "s:\(project_name =>\).*:\1 \"admin\":" $c
+    sed -i "s:\(username =>\).*:\1 \"admin\":" $c
+    sed -i "s:\(password =>\).*:\1 \"$pw\":" $c
+    sed -i "s:\(user_domain_name =>\).*:\1 \"Default\":" $c
+    sed -i "s:\(project_domain_name =>\).*:\1 \"Default\":" $c
+    sed -i "s:\(dimensions => \).*:\1 [ \"hostname\:monasca\" ]:" $c
+    start_and_enable_service  openstack-monasca-log-agent
+
+# monasca-log-agent
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: comment is wrong - monasca-agent

File: scripts/openstack-quickstart-demosetup [hunk 5]
```diff
@@ -787,6 +790,36 @@ if [ "x$with_gnocchi" = "xyes" ]; then
     #start_and_enable_service openstack-gnocchi-statsd
 fi
 
+#-----------------------------------------
+# setup Monasca
+#-----------------------------------------
+
+if [ "x$with_monasca" = "xyes" ]; then
+# monasca-log-agent
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: NIT: indent that, perhaps?

File: scripts/openstack-quickstart-demosetup (hunk 2 above)
Comment (sjamgade) [inline scripts/openstack-quickstart-demosetup]: The conf is not exactly an ini file. 
Its a custom format, made of blobs of yaml

File: scripts/openstack-quickstart-demosetup (hunk 3 above)
Comment (sjamgade) [inline scripts/openstack-quickstart-demosetup]: Done

File: scripts/openstack-quickstart-demosetup (hunk 5 above)
Comment (sjamgade) [inline scripts/openstack-quickstart-demosetup]: Done

File: scripts/openstack-quickstart-demosetup (hunk 4 above)
Comment (sjamgade) [inline scripts/openstack-quickstart-demosetup]: Done

**Comment by toabctl** [comment]
No longer needed. monasca support was backported from master

**Pull request closed without merging**